camel/toolkits/openbb_toolkit.py

```diff
@@ -0,0 +1,1126 @@
+
+    Additional data provider API keys can be configured for enhanced access:
+        - FMP (Financial Modeling Prep)
+        - Polygon.io
+        - FRED (Federal Reserve Economic Data)
+
+    Environment Variables:
+        OPENBB_PAT: OpenBB Platform Personal Access Token
+        FMP_API_KEY: Financial Modeling Prep API key
+        POLYGON_API_KEY: Polygon.io API key
+        FRED_API_KEY: FRED API key
+
+    Example:
+        >>> toolkit = OpenBBToolkit()
+        >>> df = toolkit.get_stock_quote("AAPL")
+        >>> print(df)
+    """
+
+    client: OpenBBType
+
+    @dependencies_required("openbb")
+    def __init__(self) -> None:
+        """Initialize the OpenBBToolkit.
+
+        This method sets up the OpenBB client and initializes the OpenBB
+        Hub account system.
+        """
+        from typing import cast
+
+        from openbb import obb
+
+        self.client = cast(OpenBBType, obb)
+        self._init_api_keys(silent=True)
+
+    def _init_api_keys(self, silent: bool = False) -> None:
+        """Initialize API keys from environment variables.
+
+        Args:
+            silent (bool): If True, suppress error messages.
+                Defaults to False.
+        """
+        pat = os.getenv("OPENBB_PAT")
+        if not pat:
+            pat = "dummy_key"
+
+        try:
+            self.client.account.login(pat=pat)
+
+            provider_keys = {
+                "POLYGON_API_KEY": (
+                    "polygon_api_key",
+                    "https://polygon.io",
+                ),
+                "FRED_API_KEY": (
+                    "fred_api_key",
+                    "https://fred.stlouisfed.org",
+                ),
+                "FMP_API_KEY": (
+                    "fmp_api_key",
+                    "https://financialmodelingprep.com",
+                ),
+            }
+
+            for env_var, (key_name, url) in provider_keys.items():
+                api_key = os.getenv(env_var)
+                if api_key:
+                    setattr(self.client.user.credentials, key_name, api_key)
+                elif not silent:
+                    provider = key_name.split('_')[0].upper()
+                    print(f"{env_var} not found in environment variables.")
+                    print(f"Some {provider} data may be unavailable.")
+                    print(f"Get an API key at {url}")
+                    print(f"Then set it with: export {env_var}='your-key'")
+
+        except Exception as e:
+            if not silent:
+                print(f"Failed to initialize OpenBB client: {e!s}")
+                if "Session not found" in str(e):
+                    msg = "Please make sure you have a valid "
+                    msg += "Personal Access Token (PAT)"
+                    print(msg)
+                    print("Get one at https://my.openbb.co/app/sdk")
+            raise
+
+    def search_equity(
+        self,
+        query: str,
+        provider: Literal["nasdaq", "sec"] = "nasdaq",
```

Comment:
Please link the official doc regarding what providers are supported. I checked some of their examples, I think that they support more providers than what listed here.

Reply:
Thanks @willshang76 for your mention. Just added all supported providers with the docs link to the comment.